### DISCUSSION TITLE
When hiding the navigation bar in the parent and show it in the child, it doesn't work properly when using `Store.scope` for some reason.

### DISCUSSION THREAD

**Discussion by konomae** [pointfreeco/swift-composable-architecture#806] [Q&A] [answered]
Demo Project
https://github.com/konomae/TCANavigationScopeDemo

**Works fine on iOS 15 but does not work fine on iOS14.**

I have created three examples.
https://github.com/konomae/TCANavigationScopeDemo/blob/84f310fa01d6b3cce604578e3ec7ab9e372f278f/TCANavigationScopeDemo/ContentView.swift#L117-L161

* The first is the case in question.
    * Please tap `show child`
    * (With TCA, Pass scoped store to the ChildView.)
* The second one is almost the same as the first case, but it creates a new Store.
    * Please tap `show child (without scope)`
* The third is the implementation without TCA.
    * Please tap `show child (vanilla)`

The bottom two cases work, but for some reason the top one does not. 🤔

https://user-images.githubusercontent.com/1920783/133728075-1d0da6fa-4441-44cc-9d84-80428f14a903.mp4

In the top case, the navigation bar does not appear when the screen is displayed.
However, when you move to another tab and come back again, it will be displayed.

Can someone please help me?

---

Experiment 1: Remove onAppear
https://github.com/konomae/TCANavigationScopeDemo/commit/862a158e77756171d268cfe1308d1c5db05a6962#diff-3620dff59d2cbb5235a09ac1f247d42e0445695a801ad12ac6b046abc9703c21

Works fine🤔
Navigation bar will be displayed.

https://user-images.githubusercontent.com/1920783/133729618-31555413-151b-4fd7-ba27-7d9784c5dd68.mp4

---

Experiment 2: Remove delay
https://github.com/konomae/TCANavigationScopeDemo/commit/4b56836e860ab7a8783b5fd6416fb142bafd52d0

Also works fine🤔
Navigation bar will be displayed.

**Comment by konomae** [comment, marked as answer]
This problem was caused by the parent's state change occurring when the child's action was sent, which caused the parent's View to be recreated. 😅

I was able to solve this problem by properly separating scopes. 🙌
https://github.com/konomae/TCANavigationScopeDemo/commit/088db032a8fbabb634b657f599252ca2b40b5674

**Reply by iampatbrown** [reply]
Great work @konomae Thanks for posting your solution for others :)
